File: tests/reverse/reverse_test.rs
use polyglot::reverse::reverse::iterate;
use polyglot::reverse::reverse::std;

#[test]
fn test_iterate_reverse_string() {
    for (word, want) in TESTS {
        assert_eq!(iterate(word), *want);
    }
}

#[test]
fn test_std_reverse_string() {
    for (word, want) in TESTS {
        assert_eq!(std(word), *want);
    }
}

const TESTS: &[(&'static str, &'static str)] = &[
    ("cat", "tac"),
    ("alphabet", "tebahpla"),
    ("日本語", "語本日"),
];

File: deno/tests/reverse_test.ts
import { assertEquals } from "https://deno.land/std@0.78.0/testing/asserts.ts";
import { iterate, std } from "../src/reverse.ts";

Deno.test("iterate should reverse a string", () => {
  TESTS.forEach((tc) => {
    assertEquals(iterate(tc.word), tc.want);
  });
});

Deno.test("std should reverse a string", () => {
  TESTS.forEach((tc) => {
    assertEquals(std(tc.word), tc.want);
  });
});

const TESTS = [
  { word: "cat", want: "tac" },
  { word: "alphabet", want: "tebahpla" },
  { word: "日本語", want: "語本日" },
];
